### PR TITLE
feat: basic fastapi setups & logging

### PR DESCRIPTION
**Pull request by alan910127**
---

<details open="true"><summary>Generated summary (powered by <a href="https://app.graphite.dev">Graphite</a>)</summary>

> ## TL;DR
> This pull request includes changes to the .gitignore and .vscode/settings.json files. It also adds new files and modifies existing files. The changes involve configuring a Python project, setting up logging, and adding middleware for handling HTTP requests. The code snippet enhances the Uvicorn access log format and includes various details about the client's request and response.
> 
> ## What changed
> - Changes to .gitignore and .vscode/settings.json files
> - Renaming a file, initializing a FastAPI application, setting up logging, and adding middleware for handling HTTP requests
> - Importing settings from a config file, defining a class for settings, and adding logging functionality to a Python application
> - Adding new files, modifying existing files, and importing required modules
> - Defining functions for logging and exception handling
> - Adding logging middleware function for an API
> - Enhancing Uvicorn access log format
> 
> ## How to test
> - Review the changes made to the .gitignore and .vscode/settings.json files
> - Test the functionality of the FastAPI application, including the ping endpoint and middleware for logging and correlation ID
> - Verify the logging configurations and functionality in the Python application
> - Test the logging middleware function in the API and check the access logs
> - Verify the enhanced Uvicorn access log format and check if it includes the expected details about the client's request and response
> 
> ## Why make this change
> - Update the .gitignore and .vscode/settings.json files to reflect the changes made to the project
> - Configure the Python project and set up logging functionality for better debugging and error handling
> - Add middleware for handling HTTP requests and improve the overall functionality of the API
> - Enhance the Uvicorn access log format to include more details about the client's request and response for better monitoring and troubleshooting
</details>

### CODE REVIEW

**Comment by alan910127** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #1** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/1?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #3** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/3?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #4** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/4?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #5** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/5?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #6** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/6?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/GymSquad/archive-api/1?utm_source=stack-comment).

File: .vscode/settings.json
Comment (BaBa0525) [inline .vscode/settings.json:1]: 偷抄你的 settings 來放我的 advent of code 0.0

**Review by BaBa0525 — APPROVED** [review]
好難 oao

**Comment by alan910127** [comment]
### Merge activity

* **Jan 27, 6:16 AM**: @alan910127 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/GymSquad/archive-api/1).
* **Jan 27, 6:17 AM**: @alan910127 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/GymSquad/archive-api/1).

**Pull request merged**